api/v1alpha1/warehouse_types.go

```diff
@@ -311,13 +311,22 @@ type ChartSubscription struct {
 
 // WarehouseStatus describes a Warehouse's most recently observed state.
 type WarehouseStatus struct {
+	// Conditions contains the last observations of the Warehouse's current
+	// state.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchMergeKey:"type" patchStrategy:"merge" protobuf:"bytes,9,rep,name=conditions"`
 	// LastHandledRefresh holds the value of the most recent AnnotationKeyRefresh
 	// annotation that was handled by the controller. This field can be used to
 	// determine whether the request to refresh the resource has been handled.
 	// +optional
 	LastHandledRefresh string `json:"lastHandledRefresh,omitempty" protobuf:"bytes,6,opt,name=lastHandledRefresh"`
 	// Message describes any errors that are preventing the Warehouse controller
 	// from polling repositories to discover new Freight.
+	//
+	// Deprecated: Use Conditions instead.
 	Message string `json:"message,omitempty" protobuf:"bytes,3,opt,name=message"`
```

Comment:
Can we get away with jumping straight to removing this? I don't event think we would require upgrade logic. Conditions will naturally be filled in in the normal course of business.

Reply:
Happy to just remove them. This pathway was mainly to ensure we wouldn't run into backwards compatibility issues, which I think we won't as long as the UI catches up.

Reply:
@rbreeze and I discussed last night and he'll help get the corresponding UI changes into this PR.